File: hashing/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen_nums = {}

        for i, num in enumerate(nums):
            req = target - num

            if req in seen_nums:
                return [i, seen_nums[req]]
            
            seen_nums[num] = i

        return []
        

File: hashing/two-sum.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
  public:
    vector<int> twoSum(vector<int> &nums, int target) {
      unordered_map<int, int> seenNums;

      for (int i = 0; i < nums.size(); ++i) {
        int req = target - nums[i];

        if (seenNums.count(req))
          return {i, seenNums[req]};

        seenNums[nums[i]] = i;
      }

      return {};
    }
};